web/src/server/api/routers/sessions.ts

```diff
@@ -49,16 +49,41 @@ export const sessionRouter = createTRPCRouter({
             userIds: (string | null)[] | null;
             totalCount: number;
             sessionDuration: number | null;
+            inputCost: number;
+            outputCost: number;
             totalCost: number;
+            promptTokens: number;
+            completionTokens: number;
+            totalTokens: number;
+            totalLatency: number;
           }>
         >(Prisma.sql`
-      WITH observation_metrics AS (
+      WITH trace_latencies AS (
+        SELECT
+          trace_id,
+          EXTRACT(EPOCH FROM COALESCE(MAX("end_time"), MAX("start_time"))) - EXTRACT(EPOCH FROM MIN("start_time"))::double precision AS "latency"
+        FROM
+          "observations_view"
```

Comment:
Can we fetch this from the observations table? This would have a major performance impact. Could you also quickly restructure this query the same way we did last week on the traces router? This has hughe performance improvements for our heavy users.